app/assets/javascripts/components/articles/article_graphs.jsx

```diff
@@ -1,170 +1,153 @@
-import request from '../../utils/request';
+import request from '../../utils/request.js';
 
+const ArticleGraphs = ({ article }) => {
+  const { id: article_id } = article;
 
-const ArticleGraphs = createReactClass({
-  displayName: 'ArticleGraphs',
+  const [showGraph, setShowGraph] = useState(false);
+  const [selectedRadio, setSelectedRadio] = useState('wp10_score');
+  const [articleData, setArticleData] = useState(null);
 
-  propTypes: {
-    article: PropTypes.object
-  },
+  const elementRef = useRef(null);
 
-  getInitialState() {
-    return {
-      showGraph: false,
-      selectedRadio: 'wp10_score',
-      articleData: null
+  useEffect(() => {
+    const handleClickOutside = (event) => {
+      const element = elementRef.current;
+      if (element && !element.contains(event.target)) {
+        handleHideGraph();
+      }
     };
-  },
 
-  componentDidMount() {
-    this.ref = React.createRef();
-  },
-
-  componentDidUpdate(_, prevState) {
-    if (this.state.showGraph && !prevState.showGraph) {
-      // Add event listener when the component is visible
-      document.addEventListener('mousedown', this.handleClickOutside);
-    }
-    if (!this.state.showGraph && prevState.showGraph) {
-      // remove event listener when the component is hidden
-      document.removeEventListener('mousedown', this.handleClickOutside);
-    }
-  },
+    const handlePressEscapeKey = (event) => {
+      if (event.key === 'Escape') {
+        handleHideGraph();
+      }
+    };
+    document.addEventListener('mousedown', handleClickOutside);
+    document.addEventListener('keydown', handlePressEscapeKey);
 
-  getData() {
-    if (this.state.articleData) { return; }
+    return () => {
+      document.removeEventListener('mousedown', handleClickOutside);
+      document.removeEventListener('keydown', handlePressEscapeKey);
+    };
+  }, []);
 
-    const articleId = this.props.article.id;
-    const articledataUrl = `/articles/article_data.json?article_id=${articleId}`;
+  function getData() {
+    if (articleData) {
+      return;
+    }
+    const articledataUrl = `/articles/article_data.json?article_id=${article_id}`;
     request(articledataUrl)
       .then(resp => resp.json())
       .then((data) => {
-        this.setState({ articleData: data });
+        setArticleData(data);
       });
-  },
+  }
+
+  function handleShowGraph() {
+    getData();
+    setShowGraph(true);
+  }
+
+  function handleHideGraph() {
+    setArticleData(null);
+    setShowGraph(false);
+  }
+
+  function handleRadioChange(event) {
+    setSelectedRadio(event.currentTarget.value);
+  }
+
+  function graphId() {
+    return `vega-graph-${article_id}`;
+  }
+
+  const dataIncludesWp10 = articleData?.[0]?.wp10;
 
-  handleClickOutside(event) {
-    const element = this.ref.current;
-    if (element && !element.contains(event.target)) {
-      this.hideGraph();
+  const graph = () => {
+    if (!articleData) {
+      return <Loading />;
```

Comment:
The pattern we use in the rest of the codebase is doing something like

```js
let graph;
if(...){
  graph = ...
}else{
  graph = ...
}
```

I think that's preferable to having an unnecessary function call